safehttp/form_test.go

```diff
@@ -12,1135 +12,476 @@
-			var form *safehttp.Form
-			if !strings.HasPrefix(ir.Header.Get("Content-Type"), "multipart/form-data") {
-				var err error
-				form, err = ir.PostForm()
-				if err != nil {
-					t.Fatalf(`ir.PostForm: got %v, want nil`, err)
-				}
-			} else {
-				mf, err := ir.MultipartForm(32 << 20)
-				if err != nil {
-					t.Fatalf(`ir.MultipartForm: got %v, want nil`, err)
-				}
-				form = &mf.Form
-			}
+	for _, tt := range tests {
+		t.Run(tt.val, func(t *testing.T) {
+			values := map[string][]string{"a": {tt.val}}
+			f := Form{values: values}
 
-			got := form.Int64("pizza", 0)
-			if err := form.Err(); err != nil {
-				t.Errorf(`form.Error: got %v, want nil`, err)
-			}
-			if diff := cmp.Diff(test.want, got); diff != "" {
-				t.Errorf("form.Int64:  mismatch (-want +got): \n%s", diff)
+			if got := f.Int64("a", 0); got != tt.want {
+				t.Errorf(`f.Int64("a", 0) got: %v want: %v`, got, tt.want)
 			}
-			return safehttp.Result{}
-		}))
 
-		rec := newResponseRecorder(&strings.Builder{})
-		mux.ServeHTTP(rec, test.req)
-
-		if want := 200; rec.status != want {
-			t.Errorf("response status: got %v, want %v", rec.status, want)
-		}
+			if err := f.Err(); err != nil {
+				t.Errorf("f.Err() got: %v want: nil", err)
+			}
+		})
 	}
 }
 
-func TestFormInvalidInt(t *testing.T) {
-	tests := []struct {
-		name string
-		reqs []*http.Request
-		want int64
-	}{
-		{
-			name: "Overflow integer in request",
-			reqs: []*http.Request{
-				func() *http.Request {
-					postReq := httptest.NewRequest("POST", "http://foo.com/", strings.NewReader("pizza=9223372036854775810"))
-					postReq.Header.Set("Content-Type", "application/x-www-form-urlencoded")
-					return postReq
-				}(),
-				func() *http.Request {
-					multipartReqBody := "--123\r\n" +
-						"Content-Disposition: form-data; name=\"pizza\"\r\n" +
-						"\r\n" +
-						"9223372036854775810\r\n" +
-						"--123--\r\n"
-					multipartReq := httptest.NewRequest("POST", "http://foo.com/", strings.NewReader(multipartReqBody))
-					multipartReq.Header.Set("Content-Type", `multipart/form-data; boundary="123"`)
-					return multipartReq
-				}(),
-			},
-			want: 0,
-		},
-		{
-			name: "Not an integer in request",
-			reqs: []*http.Request{
-				func() *http.Request {
-					postReq := httptest.NewRequest("POST", "http://foo.com/", strings.NewReader("pizza=diavola"))
-					postReq.Header.Set("Content-Type", "application/x-www-form-urlencoded")
-					return postReq
-
-				}(),
-				func() *http.Request {
-					multipartReqBody := "--123\r\n" +
-						"Content-Disposition: form-data; name=\"pizza\"\r\n" +
-						"\r\n" +
-						"diavola\r\n" +
-						"--123--\r\n"
-					multipartReq := httptest.NewRequest("POST", "http://foo.com/", strings.NewReader(multipartReqBody))
-					multipartReq.Header.Set("Content-Type", `multipart/form-data; boundary="123"`)
-					return multipartReq
-				}(),
-			},
-			want: 0,
-		},
+func TestFormInvalidInt64(t *testing.T) {
+	tests := []string{
```

Comment:
Please add names to this tests so it's clearer on what input the test failed, same below.